### ISSUE TITLE
NFSv4 Permission denied even with correct permissions

### ISSUE THREAD

**Issue by tailtwo** [coreos/bugs#2575]
# Issue Report #

## Bug ##

Some files hosted on a NFSv4 export with correct permissions randomly cannot be read with any user (Permission denied). A `ls -l` (emphasis on th -l) solve the problem. Maybe a `getattr` command is "fixing" it?

### Container Linux Version ###

```
$ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=2023.5.0
VERSION_ID=2023.5.0
BUILD_ID=2019-03-09-0138
PRETTY_NAME="Container Linux by CoreOS 2023.5.0 (Rhyolite)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

### Environment ###

Container Linux on bare metal and ignition config.

### Expected Behavior ###

Reading a file with correct permissions on a NFS mount returns the file content without problems.

### Actual Behavior ###

Reading a file returns `Permission denied` with any account (even root) even when the correct permissions.

### Reproduction Steps ###

  1. Mount a NFS export on the host with the following options (Kubernetes default) : `rw,relatime,vers=4.2,rsize=524288,wsize=524288,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.20.6,local_lock=none,addr=192.168.20.4`
  2. Write a file in the mounted export
  3. Issue a command that read the file, sometimes and randomly the file maybe unreadable because of a permission problem.

### Other Information ###

This behaviour appeared on 2023.5.0, the previous version (1967.6.0) worked for months without this issue. I suspect a kernel issue. I attached a tcpdump capture during a `cat` on a faulty file (sorry for the amount of packets but I couldn't narrow my scope more).

[capture.zip](https://github.com/coreos/bugs/files/3069556/capture.zip)

**Comment by ajeddeloh** [comment]
I agree this sounds like a kernel issue. Can you reproduce on the latest alpha or beta? They're running slightly newer kernels which will tell us if the fix is already upstream.

**Comment by tailtwo** [comment]
Mounted the NFS export on manjaro with kernel `5.0.5-1-MANJARO` and the problematic files are perfectly readable without any workaround. Will try CoreOS beta branch next week.

**Comment by ajeddeloh** [comment]
It's worth noting the CL will remain on the 4.19.x stable kernel branch through it's end of life. It's useful to know that it works with a 5.0.x kernel, but we won't be shipping a 5.0.x kernel.
